services/server/server.go

```diff
@@ -493,6 +496,70 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
 	return res, nil
 }
 
+// SetSSHKey uploads public key.
+func (s *Server) SetSSHKey(ctx context.Context, in *serverpb.SetSSHKeyRequest) (*serverpb.SetSSHKeyResponse, error) {
+	//TODO: validate ssh key
+	var settings *models.Settings
+	err := s.db.InTransaction(func(tx *reform.TX) error {
+		var e error
+		if settings, e = models.GetSettings(tx); e != nil {
+			return e
+		}
+
+		username := "admin"
+		usr, err := user.Lookup(username)
+		if err != nil {
+			return err
+		}
+
+		uid, err := strconv.Atoi(usr.Uid)
+		if err != nil {
+			return err
+		}
+		gid, err := strconv.Atoi(usr.Gid)
+		if err != nil {
+			return err
+		}
+
+		sshDirPath := path.Join(usr.HomeDir, ".ssh")
+		if _, err := os.Stat(sshDirPath); os.IsNotExist(err) {
+			err := os.Mkdir(sshDirPath, os.FileMode(0700))
```

Comment:
There we introduce a _third_ `err`. :)

The whole think can be simplified by using `os.MkdirAll` – it works even if path exist.

Reply:
MkdirAll creates home directory too, is it expected result? In this case we should change user owner for home directory too.

Reply:
In what case we do not have home directory?

Reply:
locally I haven't had a home directory for the some reason.

Reply:
Do we have it in current OVF and AMI?

Reply:
No